modules/gltf/editor/editor_import_blend_runner.cpp

```diff
@@ -0,0 +1,303 @@
+	Array dict_keys = p_dict.keys();
+	for (int i = 0; i < dict_keys.size(); i++) {
+		const String key = dict_keys[i];
+		String value;
+		Variant raw_value = p_dict[key];
+
+		switch (raw_value.get_type()) {
+			case Variant::Type::BOOL: {
+				value = raw_value ? "True" : "False";
+				break;
+			}
+			case Variant::Type::STRING:
+			case Variant::Type::STRING_NAME: {
+				value = raw_value;
+				value = vformat("'%s'", value.c_escape());
+				break;
+			}
+			case Variant::Type::DICTIONARY: {
+				value = dict_to_python(raw_value);
+				break;
+			}
+			default: {
+				ERR_FAIL_V_MSG("", vformat("Unhandled Variant type %s for python dictionary", Variant::get_type_name(raw_value.get_type())));
+			}
+		}
+
+		entries += vformat("'%s': %s,", key, value);
+	}
+	return vformat("{%s}", entries);
+}
+
+String dict_to_xmlrpc(const Dictionary &p_dict) {
+	String members;
+	Array dict_keys = p_dict.keys();
+	for (int i = 0; i < dict_keys.size(); i++) {
+		const String key = dict_keys[i];
+		String value;
+		Variant raw_value = p_dict[key];
+
+		switch (raw_value.get_type()) {
+			case Variant::Type::BOOL: {
+				value = vformat("<boolean>%d</boolean>", raw_value ? 1 : 0);
+				break;
+			}
+			case Variant::Type::STRING:
+			case Variant::Type::STRING_NAME: {
+				value = raw_value;
+				value = vformat("<string>%s</string>", value.xml_escape());
+				break;
+			}
+			case Variant::Type::DICTIONARY: {
+				value = dict_to_xmlrpc(raw_value);
+				break;
+			}
+			default: {
+				ERR_FAIL_V_MSG("", vformat("Unhandled Variant type %s for XMLRPC", Variant::get_type_name(raw_value.get_type())));
+			}
+		}
+
+		members += vformat("<member><name>%s</name><value>%s</value></member>", key, value);
+	}
+	return vformat("<struct>%s</struct>", members);
+}
+
+Error EditorImportBlendRunner::start_blender(const String &p_python_script, bool p_blocking) {
+	String blender_path = EDITOR_GET("filesystem/import/blender/blender3_path");
+
+#ifdef WINDOWS_ENABLED
+	blender_path = blender_path.path_join("blender.exe");
+#else
+	blender_path = blender_path.path_join("blender");
+#endif
+
+	List<String> args;
+	args.push_back("--background");
+	args.push_back("--python-expr");
+	args.push_back(p_python_script);
+
+	Error err;
+	if (p_blocking) {
+		int exitcode = 0;
+		err = OS::get_singleton()->execute(blender_path, args, nullptr, &exitcode);
+		if (exitcode != 0) {
+			return FAILED;
+		}
+	} else {
+		err = OS::get_singleton()->create_process(blender_path, args, &blender_pid);
+	}
+	return err;
+}
+
+Error EditorImportBlendRunner::do_import(const Dictionary &p_options) {
+	if (is_using_rpc()) {
+		return do_import_rpc(p_options);
+	} else {
+		return do_import_direct(p_options);
+	}
+}
+
+Error EditorImportBlendRunner::do_import_rpc(const Dictionary &p_options) {
+	kill_timer->stop();
+
+	// Start Blender if not already running.
+	if (!is_running()) {
+		// Start an XML RPC server on the given port.
+		String python = vformat("import bpy, sys, threading\n\
```

Comment:
Is there any way I could create a `xmlrpc.py` file in the modules folder and use scons to generate a C++ file containing that file's contents as a string variable? So the script doesn't have to be escaped with `\n\`.

It would make reading and maintaining this code here a lot nicer.

Reply:
I use this technique to avoid nested quoting.

```C++
static constexpr char MULTI_LINE_STRING_CONSTANT[] = R"(
// Code 1
// Code 2
)";
```